src/visualizers/VisualizerCycles.ts

```diff
@@ -0,0 +1,270 @@
+            this.sketch.square(x, y, 800 / this.width)
+
+            //Show black lines around each box
+            if (this.displayBoxLines) {
+                this.sketch.stroke(1, 1, 0)
+                this.sketch.square(x, y, 800 / this.width)
+            }
+            this.sketch.noStroke()
+        }
+    }
+
+    // draw() contains all of the mouseOver functions
+    draw() {
+        super.draw()
+
+        this.sketch.colorMode(this.sketch.HSB)
+        this.sketch.textFont('Helvetica')
+
+        //x and y-coordinate of mouse
+        const x = this.sketch.mouseX
+        const y = this.sketch.mouseY
+
+        // find the y-coordinate to set the mouseOver info
+        const tempY =
+            (800 / this.width) * Math.floor(this.numberOfTerms / this.width)
+        let lastY = 0
+        if (tempY < 720) {
+            lastY = tempY
+        }
+        if (tempY >= 720) {
+            lastY = 720
+        }
+        const squareWidth = 800 / this.width
+        const indexOfSquareMouseOver =
+            Math.floor(x / squareWidth)
+            + this.width * Math.floor(y / squareWidth)
+
+        //background large rectangle
+        this.sketch.fill(225, 50, 100)
+        this.sketch.rect(0, lastY, 800, 80)
+
+        // text settings for top of mouseOver info
+        this.sketch.fill(0, 0, 100)
+        this.sketch.textSize(25)
+
+        if (indexOfSquareMouseOver < this.workingSequence.length) {
+            let m = this.seq.getElement(
+                indexOfSquareMouseOver + this.seq.first
+            )
+            console.log(indexOfSquareMouseOver, m)
+
+            const mainText =
+                `Starting value: ${m} Cycle value: `
+                + `${collatzStepsTo1(m)} Sequence: `
+            this.sketch.text(mainText, 15, lastY + 27)
+
+            //smaller background rect for sequence
+            this.sketch.fill(225, 75, 100)
+            this.sketch.rect(18, lastY + 40, 765, 30, 10)
+
+            // text settings for the sequence list
+            this.sketch.textStyle('bold')
+            this.sketch.textSize(15)
+            this.sketch.fill(0, 0, 0)
+            this.sketch.text(m.toString(), 25, lastY + 60)
+
+            if (this.sketch.mouseIsPressed) {
+                // white outline on the square that the mouse is over
+                this.sketch.strokeWeight(3)
+                this.sketch.noFill()
+                this.sketch.stroke(0, 0, 100)
+                // draw the outline
+                this.sketch.square(
+                    squareWidth * Math.floor(x / squareWidth) + 2,
+                    squareWidth * Math.floor(y / squareWidth) + 2,
+                    800 / this.width - 4
+                )
+                this.sketch.strokeWeight(0)
+                this.sketch.fill(0, 0, 0)
+            }
+
+            // hDist tracks the horizontal distance at which the next number
+            // is placed and is dependent on the length of the previous number
```

Comment:
I'm still able to get it to make a mess with very large entries in OEIS sequences.  E.g., A007235 (for now, set it to just 7 terms so mouseover works; more terms and it seems to hang/mouseover not work).

Reply:
A007235 works for me. Kate, if you can verify it works for you, feel free to resolve this conversation.